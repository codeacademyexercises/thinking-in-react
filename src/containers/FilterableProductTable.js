import { connect } from 'react-redux';
import FilterableProductTable from '../thinking-in-react/FilterableProductTable/FilterableProductTable';
import {
    toggleStock,
    searchFor,
    } from '../actions/actions';

const mapStatetoProps = ( state )=>{
    return {
        search: state.search,
        inStock: state.inStock,
    }
}

const mapDispatchtoProps = ( dispatch )=>{
    return {
        changeSearchText: (text) => dispatch(searchFor(text)),
        toggleStock: ()=>dispatch(toggleStock()),
    }
}

export default connect(
    mapStatetoProps,
    mapDispatchtoProps,
)(FilterableProductTable);

