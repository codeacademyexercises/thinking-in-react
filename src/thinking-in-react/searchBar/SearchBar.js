import React, { Component } from 'react';
import '../App/App.css';

class SearchBar extends Component{
    render(){
        return(
            <div className="SearchBarContainer">
                <div>
                    <input type="search" placeholder="Search..." id="searchBar" defaultValue={this.props.search} onChange={this.props.SearchInput}/>
                </div>
                <div>
                    <input type="checkbox" id="inStock" defaultValue={this.props.stock} onChange={this.props.InStock}/>
                    <label >Only show products in stock</label>
                </div>
                
            </div>
        );
    }
}

export default SearchBar;