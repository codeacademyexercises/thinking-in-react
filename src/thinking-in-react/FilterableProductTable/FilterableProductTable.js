import React, { Component } from 'react';
import SearchBar from '../searchBar/SearchBar.js';
import ProductTable from '../ProductTable/ProductTable.js';

class FilterableProductTable extends Component {
    // constructor(props) {
    //     super(props);
    //     this.SearchInput = this.SearchInput.bind(this);
    //     this.InStock = this.InStock.bind(this);
    //     this.state = {search: this.props.search, 
    //                 inStock: this.props.inStock,};
    //   }
    SearchInput = (e)=>{
        // this.setState({search: e.target.value});
        this.props.changeSearchText(e.target.value);
    }
    InStock = ()=>{
        // let inStock = this.state.inStock;
        // this.setState({inStock: !inStock});
        this.props.toggleStock();
    }
  render() {
    return (
        <div>
            <SearchBar search={this.props.search} stock={this.props.inStock} SearchInput={this.SearchInput} InStock={this.InStock}/>
            <ProductTable search={this.props.search} stock={this.props.inStock}/>
        </div>
    );
  }
}

export default FilterableProductTable;
