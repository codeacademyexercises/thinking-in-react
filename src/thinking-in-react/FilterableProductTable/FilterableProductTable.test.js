import React from 'react';
import { shallow } from 'enzyme';
import FilterableProductTable from './FilterableProductTable.js';
import renderer from 'react-test-renderer';
import SearchBar from '../SearchBar/SearchBar.js';

describe('FilterableProductTable should ',()=>{
    it('render',()=>{
        const tree = renderer.create(<FilterableProductTable search="" inStock={true}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('functions working correctly and props are equal to state values',()=>{
        const wrapper = shallow(<FilterableProductTable/>);
        wrapper.setState({
            search: "",
            inStock: false,
        });
        expect(wrapper.state('search')).toEqual('');
        expect(wrapper.state('inStock')).toEqual(false);
        const instance = wrapper.instance();
        instance.SearchInput({
            target: {
                value: 'Elec',
            }
        });
        expect(wrapper.state('search')).toEqual('Elec');
        instance.InStock();
        expect(wrapper.state('inStock')).toEqual(true);
        expect(wrapper.find('SearchBar').props().search).toEqual('Elec');
        expect(wrapper.find('ProductTable').props().stock).toEqual(true);
    });
});