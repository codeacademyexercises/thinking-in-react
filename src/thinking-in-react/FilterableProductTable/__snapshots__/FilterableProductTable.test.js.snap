// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable should  render 1`] = `
<div>
  <div
    className="SearchBarContainer"
  >
    <div>
      <input
        defaultValue=""
        id="searchBar"
        onChange={[Function]}
        placeholder="Search..."
        type="search"
      />
    </div>
    <div>
      <input
        defaultValue={true}
        id="inStock"
        onChange={[Function]}
        type="checkbox"
      />
      <label>
        Only show products in stock
      </label>
    </div>
  </div>
  <div
    className="tableDetails"
  >
    <div>
      <table>
        <tbody>
          <tr>
            <th>
              Sporting Goods
            </th>
          </tr>
          <tr>
            <td>
              Football
            </td>
            <td>
              $
              49.99
            </td>
          </tr>
          <tr>
            <td>
              Baseball
            </td>
            <td>
              $
              9.99
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <table>
        <tbody>
          <tr>
            <th>
              Electronics
            </th>
          </tr>
          <tr>
            <td>
              iPod Touch
            </td>
            <td>
              $
              99.99
            </td>
          </tr>
          <tr>
            <td>
              Nexus 7
            </td>
            <td>
              $
              199.99
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
